### PR TITLE
[Improvement]: Add Paimon Table Consumer Infos In Ams Dashboard

### LINKED ISSUE #3054: [Improvement]: Add Paimon Table Consumer Infos In Ams Dashboard
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

after #3027 for paimon consumer,we need dashboard have some consumer info to know consumer paimon table what next snapshot

### How should we improve?

 image

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Liammcl**
## Why are the changes needed?

Close #3054. 

Add Paimon Table Consumer Infos In Ams Dashboard.

image

## Brief change log

-  A new Paimon Table Consumer type In Ams Dashboard.

## How was this patch tested?
* [x]  Add some test cases that check the changes thoroughly including negative and positive cases if possible
* [x]  Add screenshots for manual tests if appropriate
* [x]  Run test locally before making a pull request

## Documentation
* Does this pull request introduce a new feature? (yes)
* If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: amoro-ams/amoro-ams-dashboard/vite.config.ts [hunk 1]
```diff
@@ -95,24 +95,24 @@ export default defineConfig({
...
-    //         'Access-Control-Allow-Origin': '*',
-    //         'Access-Control-Allow-Credentials': 'true',
-    //         'Access-Control-Allow-Headers':
-    //           'Content-Type, Content-Length, Authorization, Accept, X-Requested-With , yourHeaderFeild',
-    //         'Access-Control-Allow-Methods': 'PUT,POST,GET,DELETE,OPTIONS'
-    //       }
-    //     }
-    //   }
-    // }
+     proxy: {
```
Comment (czy006) [inline amoro-ams/amoro-ams-dashboard/vite.config.ts:98]: why this proxy need to open? it just test to open it

**Review by czy006 — COMMENTED** [review]
Thanks for your first contribution, I have left a comment. cc @Liammcl

File: amoro-ams/amoro-ams-dashboard/vite.config.ts (hunk 1 above)
Comment (Liammcl) [reply amoro-ams/amoro-ams-dashboard/vite.config.ts:98]: I will fixed that!

**Comment by czy006** [comment]
I can only verify it through local tests,can you help complete the code review cc @tcodehuber

File: amoro-ams/amoro-ams-dashboard/src/language/en.ts [hunk 2]
```diff
@@ -198,8 +198,9 @@ export default {
   totalDeleteFiles: 'delete-files',
   branches: 'branches',
   tags: 'tags',
+  consumer: 'consumer',
```
Comment (tcodehuber) [inline amoro-ams/amoro-ams-dashboard/src/language/en.ts:201]: consumers: 'consumers',

File: amoro-ams/amoro-ams-dashboard/src/language/en.ts [hunk 3]
```diff
@@ -198,8 +198,9 @@ export default {
...
   nothingToShow: 'Nothing to show',
-  filterBranchesOrTags: 'Filter branches/tags',
+  filterBranchesOrTags: 'Filter branches/tags/consumer',
```
Comment (tcodehuber) [inline amoro-ams/amoro-ams-dashboard/src/language/en.ts:203]: filterBranchesOrTagsOrConsumers: 'Filter branches/tags/consumers',

Related `filterBranchesOrTags` in the code should be replaced by `filterBranchesOrTagsOrConsumers`

File: amoro-ams/amoro-ams-dashboard/src/language/zh.ts [hunk 4]
```diff
@@ -198,8 +198,9 @@ export default {
   totalDeleteFiles: '删除文件数量',
   branches: '分支',
   tags: '标签',
+  consumer: '用户',
   nothingToShow: '无内容可展示',
-  filterBranchesOrTags: '过滤分支/标签',
+  filterBranchesOrTags: '过滤分支/标签/用户',
```
Comment (tcodehuber) [inline amoro-ams/amoro-ams-dashboard/src/language/zh.ts:203]: filterBranchesOrTagsOrConsumers: '过滤分支/标签/用户',

Related filterBranchesOrTags in the code should be replaced by filterBranchesOrTagsOrConsumers

File: amoro-ams/amoro-ams-dashboard/src/views/tables/components/Selector.vue [hunk 5]
```diff
@@ -33,8 +33,14 @@ const tagSearchKey = ref<string>('')
...
 const branchList = ref<IBranchItem[]>([])
 const tagList = ref<IBranchItem[]>([])
-const actualBranchList = computed(() => branchList.value.filter(item => !branchSearchKey.value || item.label.includes(branchSearchKey.value)))
-const actualTagList = computed(() => tagList.value.filter(item => !tagSearchKey.value || item.label.includes(tagSearchKey.value)))
+const consumerList = ref<IBranchItem[]>([])
+const actualBranchList = computed(() =>branchList.value.filter((item) =>!branchSearchKey.value ||
+ item.label.includes(branchSearchKey.value)))
+const actualTagList = computed(() =>tagList.value.filter((item) => !tagSearchKey.value || 
+item.label.includes(tagSearchKey.value)))
+const actualConsumerList = computed(() =>consumerList.value.filter((item) => !tagSearchKey.value ||
```
Comment (tcodehuber) [inline amoro-ams/amoro-ams-dashboard/src/views/tables/components/Selector.vue:39]: Can we define a consumerSearchKey instead?

**Review by tcodehuber — COMMENTED** [review]
Thanks for your work, I have some comments, PTAL.

File: amoro-ams/amoro-ams-dashboard/src/views/tables/components/Selector.vue (hunk 5 above)
Comment (Liammcl) [reply amoro-ams/amoro-ams-dashboard/src/views/tables/components/Selector.vue:39]: Of course, I will fix these bugs

File: amoro-ams/amoro-ams-dashboard/src/language/zh.ts [hunk 6]
```diff
@@ -198,8 +198,9 @@ export default {
   totalDeleteFiles: '删除文件数量',
   branches: '分支',
   tags: '标签',
+  consumers: '用户',
   nothingToShow: '无内容可展示',
-  filterBranchesOrTags: '过滤分支/标签',
+  filterBranchesOrTagsOrConsumers: '过滤分支/标签/用户',
```
Comment (czy006) [inline amoro-ams/amoro-ams-dashboard/src/language/zh.ts:203]: maybe use filterBranchesOrTagsOrConsumers: '过滤分支/标签/消费者'

File: amoro-ams/amoro-ams-dashboard/src/language/en.ts [hunk 7]
```diff
@@ -198,9 +198,9 @@ export default {
   totalDeleteFiles: 'delete-files',
   branches: 'branches',
   tags: 'tags',
-  consumer: 'consumer',
+  consumers: 'consumers',
   nothingToShow: 'Nothing to show',
-  filterBranchesOrTags: 'Filter branches/tags/consumer',
+  filterBranchesOrTagsOrConsumers: 'Filter branches/tags/consumer',
```
Comment (tcodehuber) [inline amoro-ams/amoro-ams-dashboard/src/language/en.ts:203]: filterBranchesOrTagsOrConsumers: 'Filter branches/tags/consumers',

File: amoro-ams/amoro-ams-dashboard/src/types/common.type.ts [hunk 8]
```diff
@@ -359,7 +359,7 @@ export type ILineChartOriginalData = Record<string, Record<string, number>>
 export enum branchTypeMap {
   BRANCH = 'branch',
   TAG = 'tag',
-  CONSUMER = 'consumer',
+  CONSUMERS = 'consumers',
```
Comment (tcodehuber) [inline amoro-ams/amoro-ams-dashboard/src/types/common.type.ts:362]: Here you can keep Using : CONSUMER = 'consumer'

File: amoro-ams/amoro-ams-dashboard/src/views/tables/components/Selector.vue [hunk 9]
```diff
@@ -77,7 +78,7 @@ async function getTagList() {
 }
 async function getConsumerList() {
   const result = await getConsumers(props as any)
-  consumerList.value = (result.list || []).map((l: IServiceBranchItem) => ({ value: l.consumerId, label: l.consumerId, type: branchTypeMap.CONSUMER, amoroCurrentSnapshotsOfTable: l.amoroCurrentSnapshotsOfTable}))}
```
Comment (tcodehuber) [inline amoro-ams/amoro-ams-dashboard/src/views/tables/components/Selector.vue]: use `branchTypeMap.CONSUMER`

**Review by tcodehuber — APPROVED** [review]
LGTM

**Review by czy006 — APPROVED** [review]
LGTM,Thank you for your First Contribution 🎉

**Comment by Liammcl** [comment]
![image](https://github.com/user-attachments/assets/7de3d6f2-fea6-4b7b-8694-4e38ccb8a410

**Comment by Liammcl** [comment]
image

**Pull request merged**